tendermint/src/private_key.rs

```diff
@@ -36,60 +38,31 @@ impl PrivateKey {
     }
 }
 
-/// Ed25519 keypairs
-#[derive(Zeroize)]
-#[zeroize(drop)]
-pub struct Ed25519Keypair([u8; ED25519_KEYPAIR_SIZE]);
```

Comment:
I was able to replace this with ed25519-dalek's own `ed25519_dalek::Keypair` type, which validates keys.

Reply:
Yeah, this type previously didn't check that the associated point would decompress correctly (or was computed correctly from the Ed25519 "seed" in the keypair)